### DISCUSSION TITLE
I'm getting - Try signing in with a different account.

### DISCUSSION THREAD

**Discussion by onurhan1337** [nextauthjs/next-auth#4415] [Help] [answered]
### Question 💬

I am trying to sign with Github provider, but I am getting a message `Try signing in with a different account.
I did some research but couldn't find any results, would be glad if you help.

### How to reproduce ☕️

```
npx version: "8.5.0"
nextjs version: "12.1.4"
next-auth version: "^4.3.2"
react version: "^18.0.0"
```

.env file:
```
NEXTAUTH_URL=http://localhost:3000
DATABASE_URL="postgresql://secret:secret@localhost:5432/mydb?schema=secret"

GITHUB_ID=
GITHUB_SECRET=
```
and [...nextauth].ts file:

```js
import NextAuth from 'next-auth';
import { PrismaAdapter } from '@next-auth/prisma-adapter';
import { PrismaClient } from '@prisma/client';
import GitHubProvider from 'next-auth/providers/github';

const prisma = new PrismaClient();

export default NextAuth({
  adapter: PrismaAdapter(prisma),
  providers: [

    GitHubProvider({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET,
    }),
  ],

  secret: process.env.SECRET,

  pages: {
    //signIn: '/auth/signin', // Displays signin buttons
    // signOut: '/auth/signout', // Displays form with sign out button
    // error: '/auth/error', // Error code passed in query string as ?error=
    // verifyRequest: '/auth/verify-request', // Used for check email page
  },

  // Callbacks are asynchronous functions you can use to control what happens
  // when an action is performed.
  // https://next-auth.js.org/configuration/callbacks
  callbacks: {
    // async signIn(user, account, profile) { return true },
    // async redirect(url, baseUrl) { return baseUrl },
    // async session(session, user) { return session },
    // async jwt(token, user, account, profile, isNewUser) { return token }
  },

  // Events are useful for logging
  // https://next-auth.js.org/configuration/events
  events: {
    signIn: ({ user, account, profile, isNewUser }) => {
      console.log(`isNewUser: ${JSON.stringify(isNewUser)}`);
    },
    // updateUser({ user })
  },

  // Enable debug messages in the console if you are having problems
  debug: true,
});
```

when i try to sign in console error:
```bash
  provider: {
    id: 'github',
    name: 'GitHub',
    type: 'oauth',
    authorization: {
      url: 'https://github.com/login/oauth/authorize',
      params: [Object]
    },
    token: { url: 'https://github.com/login/oauth/access_token', params: {} },
    userinfo: {
      url: 'https://api.github.com/user',
      request: [AsyncFunction: request]
    },
    profile: [Function: profile],
    idToken: false,
    checks: [ 'state' ],
    clientId: '',
    clientSecret: '',
    signinUrl: 'http://localhost:3000/api/auth/signin/github',
    callbackUrl: 'http://localhost:3000/api/auth/callback/github'
  },
  message: 'client_id is required'
}
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ThangHuuVu** [comment, marked as answer]
> I am trying to sign with the Google provider

Do you mean GitHub provider?
You'll need to provide the `GITHUB_ID` and `GITHUB_SECRET` for the provider to work correctly. See how to obtain the value here: https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps

**Reply by onurhan1337** [reply]
Yeah I mean Github Provider, sorry for mistake. Thank you for your interest, I'll try and share the results here.

**Comment by onurhan1337** [comment]
@ThangHuuVu error solved. Thanks for helping.

**Reply by adamfadil17** [reply]
so, what's the solution? can you share to me

**Comment by rohann06** [comment]
i'm having a same erorr but in google provider

**Reply by razzuSharma** [reply]
Did you got the solution?

**Reply by DaronAdams** [reply]
@razzuSharma checking in to see if you got the solution as well lol

**Reply by razzuSharma** [reply]
Haha, I didn't solved it but i tried with Facebook auth too and the error was same and i came into conclusion i think i had wrong setup for prisma.

**Reply by jryebread** [reply]
i had to add a post install script to my package.json to do prisma generate when deploying

    "postinstall": "prisma generate",

**Reply by razzuSharma** [reply]
great

**Comment by debadeepsensl** [comment]
Hi, I am getting the same error with Okta provider. I have the following keys set in my `.env` file. Could you please guide me on what might be going wrong here?
```
OKTA_OAUTH2_ISSUER
OKTA_OAUTH2_CLIENT_ID
OKTA_OAUTH2_CLIENT_SECRET
```

**Comment by pmtutor** [comment]
i got the same error but with apple login......can anyone help?

**Comment by stevenwilber-deliveroo** [comment]
I had the same issue, but my .env file was in the wrong location.

**Comment by Simpleegithub** [comment]
import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";

const handler= NextAuth({
    providers: [
        GoogleProvider({
            clientId: process.env.GOOGLE_CLIENT_ID,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET,
            httpOptions: {
                timeout: 10000
            }
        }),
    ],
})

export {handler as GET, handler as POST} just add      httpOptions: {
                timeout: 10000
            } that one and your problem is solved thanks me later 🙂

**Comment by mlbrothers** [comment]
Solution for Google login for docker:

Run commands in this sequence:
```
# Use official Node.js image as a base
FROM node:20-alpine

RUN apk add --no-cache openssl

# Set the working directory in the container
WORKDIR /app

# Copy the package.json and package-lock.json files
COPY package*.json ./

# Install dependencies
RUN npm install
RUN npm i -g prisma

# Copy the rest of the application code
COPY . .
RUN npx prisma generate --schema ./prisma/schema.prisma
RUN ls -R /app

# Build the Next.js application
RUN npm run build

# Expose the application port (usually 3000 for Next.js)
EXPOSE 3000

# Start the application
CMD ["npm", "start"]
```

It will mostly solve your issue. If you have `prisma generate` in "postinstall", remove it.
